censustree/censustree.go

```diff
@@ -1,27 +1,88 @@
 package censustree
 
 import (
+	"fmt"
+	"path/filepath"
+	"strings"
 	"sync/atomic"
 
+	"github.com/vocdoni/arbo"
 	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/db/badgerdb"
 	"go.vocdoni.io/dvote/tree"
+	"go.vocdoni.io/proto/build/go/models"
+)
+
+const (
+	// TreeUnknown is the default value used for censusTree implementations
+	// which are not part of the this factory.
+	TreeTypeUnknown = models.Census_UNKNOWN
+	// TreeTypeArboBlake2b defines a tree type that uses arbo with Blake2b
+	// hash function. Is thought for being used when computation speed is
+	// important.
+	TreeTypeArboBlake2b = models.Census_ARBO_BLAKE2B
+	// TreeTypeArboPoseidon defines a tree type that uses arbo with Poseidon
+	// hash function. Is thought for being used when zkSNARK compatibility
+	// is needed (with circomlib).
+	TreeTypeArboPoseidon = models.Census_ARBO_POSEIDON
 )
 
 // CensusTree implements the Tree used for census
 type CensusTree struct {
 	*tree.Tree
 
-	public uint32
+	public     uint32
+	censusType models.Census_Type
 }
 
+// TMP to be defined the production circuit nLevels
+const nLevels = 32
+
 // New returns a new CensusTree, if there already is a CensusTree in the
 // database, it will load it.
-func New(wTx db.WriteTx, opts tree.Options) (*CensusTree, error) {
-	t, err := tree.New(wTx, opts)
+func New(wTx db.WriteTx, censusType models.Census_Type, name, storageDir string) (
+	*CensusTree, error) {
+	var hashFunc arbo.HashFunction = arbo.HashFunctionBlake2b
+
+	switch censusType {
+	case 0:
+		hashFunc = arbo.HashFunctionBlake2b
+		censusType = models.Census_ARBO_BLAKE2B
+	case models.Census_ARBO_BLAKE2B:
+		hashFunc = arbo.HashFunctionBlake2b
+	case models.Census_ARBO_POSEIDON:
```

Comment:
use your constants here? would also be nice to use a constant for 0

Reply:
In fact, I've added 'custom constants' that where an alias over the protobuff constants, updated it to not have the custom constants and to use directly the protobuff constants in this package.